File: widgets/selphid_plugin_v2/android/src/main/java/com/selphid/flutter/plugin/selphid_plugin/SelphidScanMode.java
package com.selphid.flutter.plugin.selphid_plugin;

import com.facephi.fphiselphidwidgetcore.WidgetSelphIDScanMode;

public enum SelphidScanMode {
    SM_GENERIC("GenericMode") {
        @Override
        public WidgetSelphIDScanMode getScanMode() {
            return WidgetSelphIDScanMode.SMGeneric;
        }
    },
    SM_SEARCH("SearchMode") {
        @Override
        public WidgetSelphIDScanMode getScanMode() {
            return WidgetSelphIDScanMode.SMSearch;
        }
    },
    SM_SPECIFIC("SpecificMode") {
        @Override
        public WidgetSelphIDScanMode getScanMode() { return WidgetSelphIDScanMode.SMSpecific; }
    };

    private final String name;

    SelphidScanMode(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    public abstract WidgetSelphIDScanMode getScanMode();

}

File: widgets/selphid_plugin_v2/lib/SelphIDScanMode.dart
class SelphIDScanMode {

  final _value;
  const SelphIDScanMode._internal(this._value);
  toString() => '$_value';

  static const CAP_MODE_GENERIC = const SelphIDScanMode._internal('GenericMode');
  static const CAP_MODE_SPECIFIC = const SelphIDScanMode._internal('SpecificMode');
  static const CAP_MODE_SEARCH = const SelphIDScanMode._internal('SearchMode');

}